src/language-handlebars/printer-glimmer.js

```diff
@@ -162,6 +194,46 @@ function print(path, options, print) {
       return concat([n.key, "=", path.call(print, "value")]);
     }
     case "TextNode": {
+      const inAttrNode = path.stack && path.stack.includes("attributes");
+
+      if (options.htmlWhitespaceSensitivity === "strict" && !inAttrNode) {
+        if (n.chars.match(/^\s+$/)) {
+          const lineBreaksCount = countNewLines(n.chars);
+          const breaks = new Array(lineBreaksCount || 1).fill(hardline);
+
+          if (isLastNodeOfSiblings(path)) {
+            return dedent(concat(breaks));
+          }
+
+          return concat(breaks);
+        }
+
+        const [lead] = n.chars.match(/^\s+/) || [];
+        const [tail] = n.chars.match(/\s+$/) || [];
+
+        let text = n.chars;
+        const hasLeadingWhitespace = lead && isFirstNodeOfSiblings(path);
+        const hasTrailingWhitespace = tail && isLastNodeOfSiblings(path);
+
+        let leadBreaks = [];
+        if (hasLeadingWhitespace) {
+          const leadLineBreaksCount = countNewLines(lead);
+          text = text.replace(/^\s+/, "");
```

Comment:
I don't think we should use `\s` in all these places, shouldn't it the same as html whitespace ? https://github.com/prettier/prettier/blob/196dfd207a78b11ddbfdee258ce41b55c3a3fa70/src/language-html/utils.js#L17

Reply:
this is right, thank you

Reply:
Need tests for it, you can use `snippets` to add escaped whitespace https://github.com/prettier/prettier/blob/196dfd207a78b11ddbfdee258ce41b55c3a3fa70/tests/html/whitespace/jsfmt.spec.js#L6

Reply:
excellent!